quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/update/QuarkusCli38to315UpdateIT.java

```diff
@@ -0,0 +1,362 @@
+package io.quarkus.ts.quarkus.cli.update;
+
+import static io.quarkus.test.util.QuarkusCLIUtils.QuarkusDependency;
+import static io.quarkus.test.util.QuarkusCLIUtils.addDependenciesToPom;
+import static io.quarkus.test.util.QuarkusCLIUtils.getDependencies;
+import static io.quarkus.test.util.QuarkusCLIUtils.getPom;
+import static io.quarkus.test.util.QuarkusCLIUtils.readPropertiesFile;
+import static io.quarkus.test.util.QuarkusCLIUtils.writePropertiesToPropertiesFile;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+import org.apache.http.HttpStatus;
+import org.apache.maven.artifact.versioning.DefaultArtifactVersion;
+import org.apache.maven.model.Dependency;
+import org.apache.maven.surefire.shared.lang3.tuple.Pair;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.QuarkusCliRestService;
+import io.quarkus.test.scenarios.annotations.EnabledOnQuarkusVersion;
+import io.quarkus.test.util.QuarkusCLIUtils;
+
+@Tag("quarkus-cli")
+@EnabledOnQuarkusVersion(version = "3.15.*", reason = "This only works on 3.15")
```

Comment:
When is this test going to be run? We test 999-SNAPSHOT in the main branch that you want to test this PR with. So IIUC this PR is not tested by GH CI and will never run in the main branch?

Reply:
What I am trying to ask is - is this PR 32 files of dead code or I missed something?

Reply:
You're right, this should go into 3.15 branch. I originally opened this just as a DRAFt, before 3.15 branch was available (IIRC) and forgot to change it.

Thank you for spotting this!